mujoco_warp/_src/sensor.py

```diff
@@ -495,53 +535,79 @@ def _sensor_pos(
 def sensor_pos(m: Model, d: Data):
   """Compute position-dependent sensor values."""
 
-  if (m.sensor_pos_adr.size == 0) or (m.opt.disableflags & DisableBit.SENSOR):
+  if m.opt.disableflags & DisableBit.SENSOR:
     return
 
-  wp.launch(
-    _sensor_pos,
-    dim=(d.nworld, m.sensor_pos_adr.size),
-    inputs=[
-      m.body_iquat,
-      m.jnt_qposadr,
-      m.geom_bodyid,
-      m.geom_quat,
-      m.site_bodyid,
-      m.site_quat,
-      m.cam_bodyid,
-      m.cam_quat,
-      m.cam_fovy,
-      m.cam_resolution,
-      m.cam_sensorsize,
-      m.cam_intrinsic,
-      m.sensor_type,
-      m.sensor_datatype,
-      m.sensor_objtype,
-      m.sensor_objid,
-      m.sensor_reftype,
-      m.sensor_refid,
-      m.sensor_adr,
-      m.sensor_cutoff,
-      m.sensor_pos_adr,
-      d.time,
-      d.qpos,
-      d.xpos,
-      d.xquat,
-      d.xmat,
-      d.xipos,
-      d.ximat,
-      d.geom_xpos,
-      d.geom_xmat,
-      d.site_xpos,
-      d.site_xmat,
-      d.cam_xpos,
-      d.cam_xmat,
-      d.subtree_com,
-      d.actuator_length,
-      d.ten_length,
-    ],
-    outputs=[d.sensordata],
-  )
+  if m.sensor_pos_adr.size > 0:
+    wp.launch(
+      _sensor_pos,
+      dim=(d.nworld, m.sensor_pos_adr.size),
```

Comment:
I think because dim here evaluates to `(d.nworld, 0)` Warp is smart enough to skip the kernel launch, which means you don't need the `if m.sensor_pos_adr.size > 0` guard above

Comment applies here and below